src/profile-logic/profile-data.js

```diff
@@ -1027,6 +1027,53 @@ export function filterThreadToSearchString(
   );
 }
 
+export function filterThreadByTab(
```

Comment:
Please add a documentation block to the top of this function. Also, I had a bit of trouble reading through the implementation. I think it would be a good idea to try and explain the intent of what the function is doing with some inline comments as well.